### ISSUE TITLE
printing fits_header attribute of Map object causes ValueError exception if metadata values include comments

### ISSUE THREAD

**Issue by jslavin** [sunpy/sunpy#7926]
### Describe the bug

I've been trying to add values with comments to the metadata of a Sunpy map. I can create the map with the metadata with comments, e.g. mdata['value'] = (42, 'meaning of life'). And if I print them it looks right, roughly, printing as ('value', (42, 'meaning of life')). However if I try to do mmap.fits_header, I get an error:

### To Reproduce

import sunpy

sunpy.map.Map(...)
etc

**Comment by jslavin** [comment]
It appears that Map metadata doesn't support comments. If I create a map object from data and a fits header object that has comments for some of its keywords, the comments just get dropped in the map metadata.

**Comment by Cadair** [comment]
This is just a quick drive-by from my phone, but I am pretty sure sunpy meta doesn't support the `= (value, comment)` syntax like astropy. I think you need to edit the key comments dict (I can't remember exactly where that is)

**Comment by jslavin** [comment]
Yes, I found a way to add comments to the metadata. You have to add items to the 'keycomments' dictionary:
mmap = sunpy.map.Map(fits_file)
mdata = mmap.meta
mdata["value"] = 42
mdata['keycomments"]["value"] = "meaning of life"

Whether doing that could/should be made simpler is something that could be discussed.
